src/components/h5p-button.js

```diff
@@ -56,8 +56,45 @@ H5P.Components.Button = (function () {
       H5P.Tooltip(button, { text: params.tooltip, position: params.tooltipPosition ?? 'top' });
     }
 
+    if (params.icon && H5P.Components.Button.IconOnlyObserver) {
+      H5P.Components.Button.IconOnlyObserver.observe(button);
+    }
+
     return button;
   }
 
   return Button;
 })();
+
+/** @constant {number} MAX_LABEL_LINE_COUNT Maximum number of lines for label before hiding it */ 
+/** @constant {number} MAX_LABEL_WIDTH_RATIO Maximum width ratio between label and button before hiding label */
+const MAX_LABEL_LINE_COUNT = 1;
+const MAX_LABEL_WIDTH_RATIO = 0.85;
+
+H5P.Components.Button.IconOnlyObserver = 
```

Comment:
H5P Group's eslint configuration uses this rule:
"There should be no line break before or after '=' (https://eslint.org/docs/latest/rules/operator-linebreak)", so should rather be:

```
H5P.Components.Button.IconOnlyObserver = H5P.Components.Button.IconOnlyObserver
  || new ResizeObserver(H5P.Components.utils.debounce((entries) => {
```